src/refactorings/extract/extract-variable/parts.test.ts

```diff
@@ -0,0 +1,43 @@
+import { Parts } from "./parts";
+
+import { Selection } from "../../../editor/selection";
+import { Position } from "../../../editor/position";
+
+describe("Parts", () => {
+  it("should identify different parts of a Code", () => {
+    const code = "Hello world!";
+    const selection = new Selection([0, 6], [0, 11]);
+
+    const { before, selected, after } = new Parts(code, selection);
+
+    expect(before).toBe("Hello ");
+    expect(selected).toBe("world");
+    expect(after).toBe("!");
+  });
+
+  it("should take offset into account", () => {
+    const code = "Hello world!";
+    const selection = new Selection([1, 8], [1, 13]);
+    const offset = new Position(1, 2);
+
+    const { before, selected, after } = new Parts(code, selection, offset);
+
+    expect(before).toBe("Hello ");
+    expect(selected).toBe("world");
+    expect(after).toBe("!");
+  });
+
+  // This doesn't work yet for multi-lines code because we don't support it.
+  it.skip("should work on a single-line selection of a multi-lines code", () => {
```

Comment:
This is the test that should be implemented if we want to make multi-lines selection works.